test/integration/client/cancel-query-tests.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+var helper = require('./test-helper')
+var Query = helper.pg.Query
+
+// before running this test make sure you run the script create-test-tables
+test('simple query interface', function () {
```

Comment:
This test doesn’t accept a `done` callback or return a promise, so its assertions won’t fail at the right time. The `assert(true)` can’t fail at all and doesn’t confirm that `row` is called. Both queries are queued, so the important cancellation path isn’t tested at all. (Actually, the first one is queued too, because the test doesn’t wait for the client to connect.)

Queries spliced out of the queue should probably reject/emit an error like they would if cancelled while running. Then this test can use the promise API and:

- wait for the connection to complete
- start a query, asserting that it resolves
- start two more queries, cancelling them both and asserting that both reject